File: src/main/java/conf/Initializer.java
package conf;

import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer;

public class Initializer extends AbstractAnnotationConfigDispatcherServletInitializer {

    @Override
    protected String[] getServletMappings() {
        return new String[] { "/*" }; }

    @Override
    protected Class<?>[] getRootConfigClasses() {
        return new Class[] { SecurityConfig.class };
    }


    // Dispatcher Servlet
    @Override
    protected Class<?>[] getServletConfigClasses() {
        return new Class[] { DbConfig.class, HsqlDataSource.class }; }

}

File: src/main/kotlin/config/Initializer.kt
package config

import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer

class Initializer : AbstractAnnotationConfigDispatcherServletInitializer() {
    override fun getServletMappings(): Array<String> {
        return arrayOf("/api/*")
    }

    override fun getRootConfigClasses(): Array<Class<*>> {
        return arrayOf()
    }

    override fun getServletConfigClasses(): Array<Class<*>> {
        return arrayOf(MvcConfig::class.java)
    }
}